test/pubsub.node.js

```diff
@@ -52,23 +52,26 @@ function stopTwo (nodes, callback) {
 // TODO: consider if all or some of those should come here
 describe('.pubsub', () => {
   describe('.pubsub on (default)', (done) => {
-    it('start two nodes and send one message', (done) => {
+    it('start two nodes and send one message, then unsubscribe', (done) => {
+      const data = Buffer.from('test')
+      const handler = (msg, nodes, cb) => {
+        expect(msg.data).to.eql(data)
+        cb(null, nodes)
+      }
       waterfall([
         (cb) => startTwo(cb),
         (nodes, cb) => {
-          const data = Buffer.from('test')
-          nodes[0].pubsub.subscribe('pubsub',
-            (msg) => {
-              expect(msg.data).to.eql(data)
-              cb(null, nodes)
-            },
-            (err) => {
+          nodes[0].pubsub.subscribe('pubsub', (msg, nodes, cb) => handler, (err) => {
+            expect(err).to.not.exist()
+            nodes[1].pubsub.publish('pubsub', data, (err) => {
               expect(err).to.not.exist()
-              setTimeout(() => nodes[1].pubsub.publish('pubsub', data, (err) => {
-                expect(err).to.not.exist()
-              }), 500)
-            }
-          )
+            })
+            nodes[0].pubsub.unsubscribe('pubsub', handler, (err) => {
+              expect(err).to.not.exist()
+              console.log('\tunsubscribed!')
+              done()
```

Comment:
It's possible done could be called before we run the expect in the publish handler. It would be good to include `chai-checkmark` in this file and mark each of the expects.  That way we can do `expect(4).checks(done)` so we can ensure everything is verified before finishing the test.

Remove the console.log as well and I think this is good to go. Thank you for the PR!

Reply:
Sounds good! I've added chai-checkmark and the check to make sure all the expects happen, as well as moved done and got rid of the console.log.

Reply:
When I remove `done()` at line 72, I get `Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.` If we place `done()` at line 77, at the end of the waterfall, it works. I'm not sure if this is corrrectly allows for the expect check to happen though.